### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: '/config/www/alexa_tts'

### ISSUE THREAD

**Issue by K987D** [custom-components/alexa_media_player#2032]
**Description**

With the newest version of alexa_media_player v.4.7.1 I cannot see any of my Echo Dot devices. This was my first installation and although I repeated the process over 5 times with slight variations I never got my Alexa devices to show in the integration.

![amp_issue](https://github.com/custom-components/alexa_media_player/assets/145480595/5ad44706-a93b-4a0b-9a55-c62ccf295823)

**What I did**

1. I got the newest alexa_media_player (version v. 4.7.1) via HACS on Home Assistant (2023.9.2).
2. Installed the integration and provided my credentials (e-mail-adress and password of my Amazon account / Alexa app).
4. Made sure that 2FA was activated for my Amazon account and added a new app there.
5. Said barcode is not readable and transferred the code from Amazon back to the alexa_media_player form.
6. Got the OTP from alexa_media_player, transferred it to Amazon. Success.
7. Typed in my e-mail-address and password again in alexa_media_player, then got another OTP on my phone and typed it in there.

The integration was set up but no devices would show. I did the whole process multiple times, changed "amazon.de" to "amazon.com" (I live in Germany, so .de would be correct) one time, made absolutely sure my credentials were correct. I even got a notification by my Alexa app that an iOS mobile device in my state just logged in to the app, so the connection seems to be established.

**System details**

- Home-assistant (version): 2023.9.2
- alexa_media (version from `const.py` or HA startup): v. 4.6.5
- alexapy (version from `pip show alexapy` or HA startup): 1.26.8
- Amazon [2FA is enabled]: yes

**Logs**

_WARNING_

Logger: homeassistant.core
Source: core.py:1837
First occurred: 12:36:06 (1 occurrences)
Last logged: 12:36:06
Unable to remove unknown service notify/alexa_media

_ERROR_

Logger: homeassistant.components.media_player
Source: custom_components/alexa_media/media_player.py:106
Integration: Mediaplayer (documentation, issues)
First occurred: 12:34:48 (2 occurrences)
Last logged: 12:36:41
Error while setting up alexa_media platform for media_player

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 359, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/alexa_media/media_player.py", line 158, in async_setup_entry
    if await async_setup_platform(
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/alexa_media/media_player.py", line 106, in async_setup_platform
    os.mkdir(upload_path)
FileNotFoundError: [Errno 2] No such file or directory: '/config/www/alexa_tts'

**Logs after restart**

Logger: homeassistant.components.media_player
Source: custom_components/alexa_media/media_player.py:106
Integration: Mediaplayer ([documentation](https://www.home-assistant.io/integrations/media_player), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+media_player%22))
First occurred: 13:21:47 (1 occurrences)
Last logged: 13:21:47
Error while setting up alexa_media platform for media_player

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 359, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/alexa_media/media_player.py", line 158, in async_setup_entry
    if await async_setup_platform(
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/alexa_media/media_player.py", line 106, in async_setup_platform
    os.mkdir(upload_path)
FileNotFoundError: [Errno 2] No such file or directory: '/config/www/alexa_tts'

**Comment by JoseMlCuevas** [comment]
Hi, I have the same problem after update to version v. 4.7.1 of Alexa_media in HACS
![2023-09-20](https://github.com/custom-components/alexa_media_player/assets/50666586/33110da7-cc08-4f85-83e1-480f8a9ea70d)

I uninstalled and configured it over 4 times to fix the problem, but nothing happened.

Home Assistant 2023.9.2
Supervisor 2023.09.2
Operating System 10.5
Frontend 20230911.0 - latest

**Comment by xel-x** [comment]
Try creating the directory:
/config/www/alexa_tts

Then restart.

**Comment by K987D** [comment]
Thank you @xel-x ! Appreciate your help.
That did the trick for me. Now all my Echo Dots are showing. :) (Didn't test much but putting a device in DND mode works, so I guess the rest should work normally.) Should this folder be automatically created by alexa_media_player and it just didn't do it in my case? Just wondering why this folder didn't exist.

**Comment by JoseMlCuevas** [comment]
Thank you very much @xel-x , I appreciate your time to help us, It works immediately. I never thought that could be that.

**Comment by xel-x** [comment]
Freaky I didn't invest much time into helping - simply got the same problem and found this issue - than tried to solve it myself.

I can only hope there isn't something else missing inside the folder. That's also why I can't give any insights on why the folder is missing in the first place.

**Comment by enricota66** [comment]
I had the same problem, my installation is quite clean and updated, the problem could be the amazon 2-factor authentication, but I'm not sure.

**Comment by liom69** [comment]
This works! thank you

**Comment by alexberrill** [comment]
Thank you so much. This was driving me nuts but works now! Now I just have to go in and update all of my automations :|

**Comment by Bclemm** [comment]
New to this, how do I create the directory?

**Comment by enricota66** [comment]
> New to this, how do I create the directory?

If you are on Home assistant  OS, for example you can install one file editor on Sistem/Components.  And use that for make a new directory on config folder.

**Comment by alexberrill** [comment]
@Bclemm using file editor create a folder in the config folder called “www”. Within that new folder create another folder called “alexa_tts”. Once done restart HA and you should see your devices. No idea why this works but it does :) thanks again @xel-x

**Comment by alexberrill** [comment]
Now I need to figure out why “MyQ” and “Eufy Robovacs” have stopped working for me. I love HA but its so annoying that things suddenly stop working and you have to hunt for solutions particularly after updates.

**Comment by 007-Ninja** [comment]
Thanks, I've been struggling with this since I upgraded to HACS 4.71 from 6...something.  Notify no longer existed for alexa, could not find media players in my devices, cloud say didn't work.  Added this www/alexa_tts directory and now I am getting alexa to talk again.  I imagine I will need to tweek my TTS automations.  But I am fine with that so long as alexa still talks to me when I trigger her.
